bottlerocket/agents/src/bin/eks-resource-agent/eks_provider.rs

```diff
@@ -128,6 +131,167 @@ enum ClusterConfig {
         region: String,
     },
 }
+#[derive(Serialize, Debug, EnumString)]
+enum IPFamily {
+    IPv6,
+    IPv4,
+}
+
+/// Configuration for setting up an EKS cluster using eksctl yaml file.
+///
+/// # Fields:
+/// - `api_version`: Specifies the version of the EKS configuration API.
+/// - `kind`: Indicates the type of the configuration, typically "ClusterConfig".
+/// - `metadata`: Metadata about the cluster for instance name, region, and version.
+/// - `availability_zones`: List of availability zones where cluster should be deployed.
+/// - `kubernetes_network_config`: Configuration for the Kubernetes network: IPv6 or IPv4.
+/// - `addons`: List of EKS addons to be enabled on the cluster.
+/// - `iam`: IAM configuration, especially for OIDC.
+/// - `managed_node_groups`: List of managed node groups for the cluster.
+#[derive(Serialize)]
+#[serde(rename_all = "camelCase")]
+struct EksctlYamlConfig {
+    /// Version of the EKS configuration API
+    api_version: String,
+    /// Type of the configuration, typically "ClusterConfig".
+    kind: String,
+    /// Metadata about the cluster for instance name, region, and version.
+    metadata: Metadata,
```

Comment:
nit: Metadata is a pretty generic term. We can name the field and struct whatever we want and have serde rename it to the proper one for the file format. We don't have to be limited here by the names other things have chosen if it helps clarify things. If we ever wanted our *own* metadata for our EKS provider, the naming might be confusing. 
```suggestion
    #[serde(rename = "metadata")]
    eksctl_metadata: EksctlMetadata,
```